Mathlib/Algebra/Group/Subsemigroup/Operations.lean

```diff
@@ -595,6 +595,9 @@ theorem coe_srange (f : M →ₙ* N) : (f.srange : Set N) = Set.range f :=
 theorem mem_srange {f : M →ₙ* N} {y : N} : y ∈ f.srange ↔ ∃ x, f x = y :=
   Iff.rfl
 
+@[to_additive] theorem mem_srange_mk {f : M → N} (hf) {y : N} :
+    y ∈ srange ⟨f, hf⟩ ↔ y ∈ Set.range f := Iff.rfl
```

Comment:
```suggestion
@[to_additive (attr := simp)] theorem srange_mk (f : M → N) (hf) :
    srange ⟨f, hf⟩ = ⟨Set.range f, _⟩ := rfl
```
seems more appropriate. See eg [`Prod.mk_add_mk`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Group/Prod.html#Prod.mk_add_mk)

Reply:
I added this, but it doesn't work in `Monic.supDegree_mul_of_ne_zero`, where we're rewriting backwards to prove `supDegree D p + supDegree D q ∈ Set.range D`

Reply:
Alternative proof was manually doing
```lean
  · have hp' := supDegree_mem_range D hp
    have hq' := supDegree_mem_range D hq.ne_zero
    rw [Set.mem_range] at *
    obtain ⟨p', hp'⟩ := hp'
    obtain ⟨q', hq'⟩ := hq'
    use p' + q'
    rw [hadd, hp', hq']
```

Reply:
Fixed